src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -400,6 +556,13 @@ private static byte[] SendUDPRequest(string browserHostname, int port, byte[] re
 
             const int sendTimeOut = 1000;
             const int receiveTimeOut = 1000;
+
+            IPAddress address = null;
+            if (IPAddress.TryParse(browserHostname, out address) && address.AddressFamily == AddressFamily.InterNetworkV6)
```

Comment:
Why not just set UdpClient to use the address's family if address != null, and then pass the (IPv6) hostName? Is there an issue with passing IPv6 addresses to send async?